pkg/performanceprofile/controller/performanceprofile/hypershift/hypershift.go

```diff
@@ -33,7 +33,8 @@ type ControlPlaneClientImpl struct {
 	// on which the control-plane objects reside
 	hostedControlPlaneNamespaceName string
 
-	opts controlPlaneClientImplOptions
+	yamlSerializer *serializer.Serializer
```

Comment:
ok, but why? performances?

Reply:
Spare the need of recreating the serializer on every call

Reply:
the tradeoff here is between reusing objects, which can possibly have unexpected state, vs recreating every time. The latter seems cheap enough, but the serializer should be stateless, so I'm ok with this change, even though I'm skeptical about the gains here.